### PR TITLE
Updates and fixes to tests

### PR DESCRIPTION
**Pull request by jentfoo**
I recently looked at this project and it looked like `./test` failed to complete.  These updates are to fix the tests in this project.

### CODE REVIEW

File: test [hunk 1]
```diff
@@ -11,6 +11,9 @@ LINTABLE=$( go list -tags=golint -f '
 
 go test -v -i -race github.com/coreos/go-oidc/...
```
Comment (zmb3) [inline test:12]: Remove the first one?

File: test [hunk 2]
```diff
@@ -11,6 +11,9 @@ LINTABLE=$( go list -tags=golint -f '
...
 go test -v -race github.com/coreos/go-oidc/...
-golint $LINTABLE
+if ! golint $LINTABLE &> /dev/null
+then
+  echo "golint not installed, skipping linting"
```
Comment (zmb3) [inline test:16]: golint is long deprecated, why are we adding it here?

https://github.com/golang/go/issues/38968

File: go.mod [hunk 3]
```diff
@@ -17,8 +17,10 @@ require (
 	github.com/golang/protobuf v1.5.2 // indirect
 	github.com/sirupsen/logrus v1.8.1 // indirect
 	golang.org/x/crypto v0.0.0-20220126234351-aa10faf2a1f8 // indirect
-	golang.org/x/sys v0.3.0 // indirect
-	golang.org/x/term v0.3.0 // indirect
+	golang.org/x/lint v0.0.0-20210508222113-6edffad5e616 // indirect
```
Comment (zmb3) [inline go.mod:20]: This looks like a mistake.

File: test (hunk 2 above)
Comment (jentfoo) [reply test:16]: I was not adding it, I was just trying to do a minimal change here to get it working.  I felt like making it optional was a good first step.

File: go.mod (hunk 3 above)
Comment (jentfoo) [reply go.mod]: Thank you, reverted

File: test (hunk 2 above)
Comment (zmb3) [reply test:16]: Gotcha

File: test (hunk 1 above)
Comment (zmb3) [reply test:12]: I assume it had something to do with the `-i` flag, which is deprecated. I would just remove that one.

Comment (jentfoo) [reply test:12]: Done, thank you!

**Review by zmb3 — APPROVED** [review]

**Pull request merged**